### PR TITLE
Role api viewset, serializer and filterset

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes: #1063 
# What's Changed

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/api/urls.py [hunk 1]
```diff
@@ -79,5 +79,8 @@
 # Webhooks
 router.register("webhooks", views.WebhooksViewSet)
 
+# Roles
```
Comment (jathanism) [inline nautobot/extras/api/urls.py:82]: Can you please add this in alphabetic order above Secrets on line 68?

File: nautobot/extras/api/views.py [hunk 2]
```diff
@@ -1170,3 +1171,14 @@ class WebhooksViewSet(ModelViewSet, NotesViewSetMixin):
     queryset = Webhook.objects.all()
     serializer_class = serializers.WebhookSerializer
     filterset_class = filters.WebhookFilterSet
+
+
+#
```
Comment (jathanism) [inline nautobot/extras/api/views.py:1176]: Same here, if you could please move Roles above Statuses alphabetically?

File: nautobot/extras/filters.py [hunk 3]
```diff
@@ -1316,3 +1317,35 @@ class Meta:
             "type_update",
             "type_delete",
         ]
+
+
+#
```
Comment (jathanism) [inline nautobot/extras/filters.py:1322]: Alpha order please like the others! 🔤

File: nautobot/extras/api/serializers.py [hunk 4]
```diff
@@ -61,7 +62,7 @@
     Tag,
     Webhook,
 )
-from nautobot.extras.utils import ChangeLoggedModelsQuery, FeatureQuery, TaggableClassesQuery
+from nautobot.extras.utils import ChangeLoggedModelsQuery, FeatureQuery, TaggableClassesQuery, RoleModelsQuery
```
Comment (timizuoebideri1) [inline nautobot/extras/api/serializers.py:65]:
```suggestion
from nautobot.extras.utils import ChangeLoggedModelsQuery, FeatureQuery, RoleModelsQuery, TaggableClassesQuery
```

**Review by glennmatthews — COMMENTED** [review]
Planning to include testing as part of this PR or a separate one?

**Comment by timizuoebideri1** [comment]
> Planning to include testing as part of this PR or a separate one?

separate PR

File: nautobot/extras/api/serializers.py [hunk 5]
```diff
@@ -1317,3 +1318,28 @@ def validate(self, data):
...
+    )
+
+    class Meta:
+        model = Role
+        fields = [
+            "url",
+            "content_types",
+            "name",
+            "slug",
+            "color",
```
Comment (glennmatthews) [inline nautobot/extras/api/serializers.py:1344]: `weight`?

File: nautobot/extras/api/serializers.py [hunk 6]
```diff
@@ -1317,3 +1318,28 @@ def validate(self, data):
...
 
         return validated_data
+
+
+#
+# Roles
+#
+
+
+class RoleSerializer(NautobotModelSerializer):
```
[the remark this answers is not in the document]
Comment (timizuoebideri1) [reply nautobot/extras/api/serializers.py:1328]: It is created in [2911](https://github.com/nautobot/nautobot/pull/2911) PR

File: nautobot/extras/api/serializers.py [hunk 7]
```diff
@@ -1317,3 +1318,28 @@ def validate(self, data):
             raise serializers.ValidationError(conflicts)
 
         return validated_data
+
```
[the remark this answers is not in the document]
Comment (timizuoebideri1) [reply nautobot/extras/api/serializers.py:1321]: It is created in [2911](https://github.com/nautobot/nautobot/pull/2911) PR

File: nautobot/extras/api/serializers.py [hunk 8]
```diff
@@ -1317,3 +1318,28 @@ def validate(self, data):
...
+
+#
```
[the remark this answers is not in the document]
Comment (timizuoebideri1) [reply nautobot/extras/api/serializers.py:1323]: It is created in [2911](https://github.com/nautobot/nautobot/pull/2911) PR

**Pull request merged**